shared/utils/combine-ovals.py

```diff
@@ -253,21 +254,28 @@ def checks(product, oval_dirs):
     already_loaded = set()
 
     for oval_dir in reversed_dirs:
-        # sort the files to make output deterministic
-        for filename in sorted(os.listdir(oval_dir)):
-            if filename.endswith(".xml"):
-
-                # skip file if we already have one with better priority
-                if filename in already_loaded:
-                    continue
-
-                with open(os.path.join(oval_dir, filename), 'r') as xml_file:
-                    xml_content = xml_file.read()
-                    if check_is_applicable_for_product(xml_content, product):
-                        body += xml_content
-                        included_checks_count += 1
-                        already_loaded.add(filename)
-
+        try:
+            # sort the files to make output deterministic
+            for filename in sorted(os.listdir(oval_dir)):
+                if filename.endswith(".xml"):
+
+                    # skip file if we already have one with better priority
+                    if filename in already_loaded:
+                        continue
+
+                    with open(os.path.join(oval_dir, filename), 'r') as xml_file:
```

Comment:
The try/except block should be around this, not the whole loop. This will skip existing dirs once any nonexisting dir is encountered!